### PR TITLE
Added certificates

### PR DESCRIPTION
**Pull request by giocalitri**
#### What's this PR do?
Adds a client for the user certificates

#### Where should the reviewer start?
`edx_api/certificates/__init__.py`
then
`edx_api/certificates/models.py`
then
`edx_api/client.py`
then
`edx_api/integration_test.py`

#### How should this be manually tested?
you need to have an instance of devstack running the branch mitocw:gdm_certificates_230
then you need to create a certificate for a course for a user.

Or simply make sure that the integration test run

#### Any background context you want to provide?
the PR on the edx side is close to be merged, but still there... you know how it works :-)

### CODE REVIEW

File: edx_api/certificates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        Returns:
+            Certificate: object representing the student certificate for a course
+        """
+        # the request is done in behalf of the current logged in user
+        resp = self.requester.get(
+            urljoin(
+                self.base_url,
+                '/api/certificates/v0/certificates/{username}/courses/{course_key}/'.format(
+                    username=username,
+                    course_key=course_id
```
Comment (noisecapella) [inline edx_api/certificates/__init__.py]: Do we need to do any escaping of the course key or username? Would either ever have a slash?

File: edx_api/certificates/models_test.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        assert len(all_certs) == 2
+        for cert in all_certs:
+            assert isinstance(cert, Certificate)
+
+    def test_all_verified_cert(self):
+        """Test for all_verified_certs"""
+        all_certs = self.certificates.all_verified_certs
+        assert len(all_certs) == 1
+        for cert in all_certs:
+            assert isinstance(cert, Certificate)
```
Comment (noisecapella) [inline edx_api/certificates/models_test.py]: Can you assert the certificate id too?

File: edx_api/certificates/__init__.py (hunk 1 above)
Comment (giocalitri) [inline edx_api/certificates/__init__.py]: it might have a `/`, but the regex should work anyway without escaping

Comment (giocalitri) [inline edx_api/certificates/__init__.py]: there are other cases we did the same (meaning not escaping the course id)

**Comment by noisecapella** [comment]
Code looks fine, it will be a short while before the sandbox I requested from @bdero is up and running to run the integration tests. By the way I noticed that a few classes (including `UserCertificate`) are not covered by code coverage. Are these meant to be tested only in the integration tests?

**Comment by giocalitri** [comment]
the integration tests take care of the missing coverage: it is basically the same for all the different packages of this library

File: edx_api/certificates/models_test.py (hunk 2 above)
Comment (giocalitri) [inline edx_api/certificates/models_test.py]: sure

**Comment by noisecapella** [comment]
Functionality tested, looks good :+1:

**Pull request merged**